From 7627c7f8a0fc0a2ec486fa92b4d32c1ce92b69aa Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 20 Jan 2013 16:22:39 +0000
Subject: [PATCH 1/2] MinGW-w64 compilation fix for CompileTimeAssert

Without -Wno-unused, on MinGW-w64 GCC 4.7.2, instances of
 CompileTimeAssert end up making being variables.

Change-Id: I0477ed2e8e174bc5e089b7b8cc3e40590f1d3f73
---
 src/lib/tools/fileinfo.cpp     | 2 +-
 src/lib/tools/filetime.h       | 6 ++++++
 src/lib/tools/filetime_win.cpp | 2 +-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/lib/tools/fileinfo.cpp b/src/lib/tools/fileinfo.cpp
index 8940a68..be4d928 100644
--- a/src/lib/tools/fileinfo.cpp
+++ b/src/lib/tools/fileinfo.cpp
@@ -205,7 +205,7 @@ template<> struct CompileTimeAssert<true> {};
 
 FileInfo::FileInfo(const QString &fileName)
 {
-    static CompileTimeAssert<
+    static ATTRIBUTE_UNUSED CompileTimeAssert<
         sizeof(FileInfo::InternalStatType) == sizeof(WIN32_FILE_ATTRIBUTE_DATA)
             > internal_type_has_wrong_size;
     if (!GetFileAttributesEx(reinterpret_cast<const WCHAR*>(fileName.utf16()),
diff --git a/src/lib/tools/filetime.h b/src/lib/tools/filetime.h
index d4e8b89..17026fd 100644
--- a/src/lib/tools/filetime.h
+++ b/src/lib/tools/filetime.h
@@ -37,6 +37,12 @@
 #include <time.h>
 #endif
 
+#if defined(__GNUC__) && !defined(COMPILER_ICC)
+#define ATTRIBUTE_UNUSED __attribute__((unused))
+#else
+#define ATTRIBUTE_UNUSED
+#endif
+
 namespace qbs {
 namespace Internal {
 
diff --git a/src/lib/tools/filetime_win.cpp b/src/lib/tools/filetime_win.cpp
index 2857393..48c14eb 100644
--- a/src/lib/tools/filetime_win.cpp
+++ b/src/lib/tools/filetime_win.cpp
@@ -40,7 +40,7 @@ namespace Internal {
 
 template<bool> struct CompileTimeAssert;
 template<> struct CompileTimeAssert<true> {};
-static CompileTimeAssert<sizeof(FileTime::InternalType) == sizeof(FILETIME)> internal_type_has_wrong_size;
+static ATTRIBUTE_UNUSED CompileTimeAssert<sizeof(FileTime::InternalType) == sizeof(FILETIME)> internal_type_has_wrong_size;
 
 FileTime::FileTime()
     : m_fileTime(0)
-- 
1.7.11.msysgit.1

